### PR TITLE
Add integration tests for Appends

### PR DESCRIPTION
**Pull request by maytasm**
Add integration tests for Appends

### Description

The native batch ingestion currently allows appending dynamically partitioned segments regardless of the original partitioning scheme. The new integration tests include the followings:

- Append dynamically-partitioned segments to a dynamically-partitioned datasource
- Append dynamically-partitioned segments to a hash-partitioned datasource. Compact hash-partitioned segments and dynamically-partitioned segments after append.
- Append dynamically-partitioned segments to a range-partitioned datasource. Compact range-partitioned segments and dynamically-partitioned segments after append.

Some notes on the tests:
- Note that the intervals of the appended segment overlap with those of original segments. This is done by using the same data for initial write and append.
- Batch append itself is pretty well covered by unit tests (the unit tests mimic the distributed ingestion using multi-threaded ingestion testing framework (see AbstractParallelIndexSupervisorTaskTest). The integration tests focus on missing parts such as querying segments of mixed partitioning schemes.

Note: This PR also updates Integration Test's areSegmentsLoaded/getLoadStatus to use the new API to verify a datasource has the latest ingested data (https://github.com/apache/druid/pull/9965) instead of a Cluster wide loadstatus API. There are two benefits. First, we are now able to query status of a specific datasource, allowing multiple integration tests to be run concurrently. Second, we are able to take advantage of the new API's `forceMetadataRefresh` to make sure that we get latest published segment information (and compare/verify against the latest published segment information). This allows us to verify the segment count after the task finished. (Note that because of this, ITAutoCompactionTest.java test was also updated to verify segment count of only used and non-overshadowed segments).  

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITAppendBatchIndexTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import java.util.function.Function;
+
+@Test(groups = {TestNGGroup.OTHER_INDEX})
+@Guice(moduleFactory = DruidTestModuleFactory.class)
+public class ITAppendBatchIndexTest extends AbstractITBatchIndexTest
+{
+  private static final Logger LOG = new Logger(ITAppendBatchIndexTest.class);
+  private static final String INDEX_TASK = "/indexer/wikipedia_local_input_source_index_task.json";
+  private static final String INDEX_QUERIES_INITIAL_INGESTION_RESOURCE = "/indexer/wikipedia_index_queries.json";
+  private static final String INDEX_QUERIES_POST_APPEND_PRE_COMPACT_RESOURCE = "/indexer/wikipedia_double_without_roll_up_index_queries.json";
```
Comment (jihoonson) [inline integration-tests/src/test/java/org/apache/druid/tests/indexer/ITAppendBatchIndexTest.java:54]: Maybe `non_perfect_rollup` is a better name than `without_roll_up`.

Comment (jihoonson) [reply integration-tests/src/test/java/org/apache/druid/tests/indexer/ITAppendBatchIndexTest.java:54]: Also, would you please add some comments for each queries about what non-perfect-rollup or perfect-rollup mean?

Comment (maytasm) [reply integration-tests/src/test/java/org/apache/druid/tests/indexer/ITAppendBatchIndexTest.java]: Done

**Comment by jihoonson** [comment]
LGTM 👍

**Pull request merged**